mdit_py_plugins/myst_blocks/index.py

```diff
@@ -40,23 +41,30 @@ def line_comment(state: StateBlock, startLine: int, endLine: int, silent: bool):
     if state.sCount[startLine] - state.blkIndent >= 4:
         return False
 
-    marker = state.srcCharCode[pos]
-    pos += 1
-
-    # Check block marker /* % */
-    if marker != 0x25:
+    if state.src[pos] != "%":
```

Comment:
Was there a reason to move away from comparing the character code?

Reply:
No reason other than that this is more self-documenting (requires no comment to explain the code) and `src` is shorter to write than `srcCharCode`. The result should be the same either way AFAICT.

Reply:
In fact I can't see the reason why `int` character codes are used so extensively in JS upstream. I'm interested though, should maybe ask the upstream devs... or can you think of a reason?

The way I see it, a `str` character is just a better/more intuitive interface for a Unicode character (an int in the range `{0...max_unicode_char}`). A `str` character supports comparisons like `"a" < char < "z"` just like an `int`, in both JS and Python, so that can't be the reason. So then I'm a bit clueless. I feel like by using `str` characters we'd avoid dozens of comments explaining the character codes, and also would never need to do the `srcCharCode` conversions (perhaps improved performance?).

Reply:
> should maybe ask the upstream devs... or can you think of a reason?

Not particularly (maybe to do with encodings?), but yeh I would try asking upstream

Reply:
> maybe to do with encodings

This I also thought of: that maybe the `int` is used to represent a Unicode codepoint whereas a JS character is a UTF-16 code unit. But no, I don't think that's the case, because the `int` is gotten with `charCodeAt` which maps to UTF-16 code unit.

Reply:
We got the answer: this is just a JS specific optimization. So IMO doesn't make sense to copy that style unless we want to match upstream code one-to-one (which isn't the case here, since we aren't porting anything).

`srcCharCode` was added as on optimization for the Python port. My gut says the best optimization for us would be to remove it entirely and use the `str` units as character codes.

Reply:
ok cheers (the issue answer was a bit vague, but I'll take it on faith)
it fine for here then cheers, but no I'm not going to start removing `srcCharCode`